src/commands/heroku.rs

```diff
@@ -0,0 +1,57 @@
+use heroku_rs::client::{Executor, Heroku};
+
+use serde::Deserialize;
+
+use serenity::framework::standard::{macros::command, Args, CommandResult};
+use serenity::model::prelude::*;
+use serenity::prelude::*;
+
+use crate::config::config::Config;
+
+#[derive(Debug, Deserialize)]
+struct HerokuApp {
+    id: String,
+    name: String,
+}
+
+#[command]
+pub fn get_apps(ctx: &mut Context, msg: &Message, _args: Args) -> CommandResult {
+    let ctx_clone = ctx.clone();
+    let data = ctx_clone.data.read();
+    let config = data.get::<Config>().expect("Expected config");
+
+    let response = heroku_client(&config.heroku_api_key)
+        .get()
+        .apps()
+        .execute::<Vec<HerokuApp>>();
+    let mut processed_app_list: Vec<HerokuApp> = Vec::new();
+
+    match response {
+        Ok((_headers, _status, json)) => {
+            if let Some(mut json) = json {
+                processed_app_list.append(&mut json);
```

Comment:
Is there a reason why you don't just use the `Vec` in the `json` variable instead of moving its elements to another one?

Reply:
I was having some trouble with persisting the json variable  out of the loop (and then passing it to msg.reply). Going to see if I can find a more elegant solution.

Reply:
I just did a small refactor where I

* [created the processed_app_list variable outside of the match and if expression](https://github.com/rust-lang/crates-io-ops-bot/blob/ad0c92d1672af3c015f49ac4e82776ebd8407ba5/src/commands/heroku.rs#L28)
* within the match and if expression, [assign the processed_list_variable to the returned json](https://github.com/rust-lang/crates-io-ops-bot/blob/ad0c92d1672af3c015f49ac4e82776ebd8407ba5/src/commands/heroku.rs#L33)
* outside of the match and if expression [pass the processed_app_list to the app_response function, then pass the result of that function to the msg.reply funtion](https://github.com/rust-lang/crates-io-ops-bot/blob/ad0c92d1672af3c015f49ac4e82776ebd8407ba5/src/commands/heroku.rs#L40)

Is there a better way to do this? I'm completely open to refining it further!

Reply:
Personally I'd replace the match and the reply with:

```rust
msg.reply(ctx, match response {
    Ok((_, _, Some(apps))) => app_response(apps),
    Ok((_, _, None)) => "You have no Heroku apps".into(),
    Err(err) => {
        println!("Err {}", err);
        "An error occured while fetching your Heroku apps".into()
    }
})?;